### PR TITLE
PUBDEV-7720 fix TestBestPracticeExample failing on multinode

### PR DESCRIPTION
**Pull request by koniecsveta**
this is the same issue as https://github.com/h2oai/h2o-3/pull/4894 but in testBestPracticeExample(), I need to update fr by DKV.put(fr) after making changes on it.

### CODE REVIEW

File: h2o-algos/src/test/java/hex/rulefit/RuleFitTest.java [hunk 1]
```diff
@@ -41,6 +41,7 @@ public void testBestPracticeExample() {
             fr.remove("boat").remove();
             fr.remove("body").remove();
             fr.remove("home.dest").remove();
+            DKV.put(fr);
```
Comment (honzasterba) [inline h2o-algos/src/test/java/hex/rulefit/RuleFitTest.java:44]: What does this fix? asFactor already does dkv.put, other calls should not require anything new to be put into DKV

Comment (koniecsveta) [reply h2o-algos/src/test/java/hex/rulefit/RuleFitTest.java:44]: when this is not here and I run on 4 nodes I get this NPE:

`java.lang.NullPointerException
	at water.fvec.Frame.<init>(Frame.java:170)
	at hex.ModelBuilder.init(ModelBuilder.java:1196)
	at hex.tree.SharedTree.init(SharedTree.java:117)
	at hex.tree.drf.DRF.init(DRF.java:53)
	at hex.tree.drf.DRF.<init>(DRF.java:36)
	at hex.rulefit.RuleFit$RuleFitDriver.trainTreeModel(RuleFit.java:289)
	at hex.rulefit.RuleFit$RuleFitDriver.computeImpl(RuleFit.java:191)
	at hex.ModelBuilder$Driver.compute2(ModelBuilder.java:238)
	at water.H2O$H2OCountedCompleter.compute(H2O.java:1563)
	at jsr166y.CountedCompleter.exec(CountedCompleter.java:468)
	at jsr166y.ForkJoinTask.doExec(ForkJoinTask.java:263)
	at jsr166y.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:974)
	at jsr166y.ForkJoinPool.runWorker(ForkJoinPool.java:1477)
	at jsr166y.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:104)`

at at hex.ModelBuilder.init(ModelBuilder.java:1196):

`setTrain(new Frame(null /* not putting this into KV */, tr._names.clone(), tr.vecs().clone()))`

some vecs in tr.vecs() are null and then this produces NPE in new Frame(...) at:

` for( Vec vec : vecs ) DKV.prefetch(vec._key);`

when you do DKV.prefetch(null)

**Review by honzasterba — APPROVED** [review]

**Comment by satai** [comment]
LGTM + I would personally remove the double Scope insertion.

**Comment by koniecsveta** [comment]
> LGTM + I would personally remove the double Scope insertion.

0034965

**Pull request merged**